src/AutoRest.CSharp/Common/Output/Models/Shared/Parameter.cs

```diff
@@ -16,7 +16,7 @@
 
 namespace AutoRest.CSharp.Output.Models.Shared
 {
-    internal record Parameter(string Name, FormattableString? Description, CSharpType Type, Constant? DefaultValue, ValidationType Validation, FormattableString? Initializer, bool IsApiVersionParameter = false, bool IsResourceIdentifier = false, bool SkipUrlEncoding = false, RequestLocation RequestLocation = RequestLocation.None, SerializationFormat SerializationFormat = SerializationFormat.Default, bool IsPropertyBag = false)
+    internal record Parameter(string Name, FormattableString? Description, CSharpType Type, Constant? DefaultValue, ValidationType Validation, FormattableString? Initializer, bool IsApiVersionParameter = false, bool IsResourceIdentifier = false, bool SkipUrlEncoding = false, RequestLocation RequestLocation = RequestLocation.None, SerializationFormat SerializationFormat = SerializationFormat.Default, bool IsPropertyBag = false, string? DefaultName = null)
```

Comment:
well I do not think it is a good name here `DefaultName`. Should it be `NameInRequest` or `WireName` or something similar?
Looks like this is only to fix the issue in sample generator that if we renamed the parameter to be a new name instead of the name in its spec, the sample generator will not be able to find which parameter to use.

Theoretically we should need this name in request when building the rest operation methods (for instance when the parameter is a query), but we did that by querying the `NameInRequest` property in `InputParameter` instead of referencing `Parameter`.
But I believe it might be nice we have that here, and the sample generator could query parameters using this property `NameInRequest` to avoid the issues caused by renamings

Reply:
sure, I can rename this property.

You are correct, this is to handle generating the samples correctly when a sample uses a custom model. It seems the lookup to find the example value for a parameter of model type uses the "original" model name vs the custom model. Though I'm not sure I entirely follow the last part of you comment, are you suggesting I change how the example values are looked up in the sample generator ?